### PR TITLE
Nullable annotations for projects using XamlX

### PR DESCRIPTION
**Pull request by MrJul**
## What does the pull request do?

This PR enables nullable reference types in projects using XamlX:
 - `Build.Tasks`
 - `Designer.HostApp`
 - `Generators`
 - `Markup.Xaml.Loader`

Depends on https://github.com/kekekeks/XamlX/pull/121

## Notes

The `Generators` project directly uses `<Nullable>enable</Nullable>` instead of including `NullableEnable.props`. This props file disables nullable warnings for `netstandard2.0` as they're usually handled by the `net8.0` build, which isn't targeted by this project.

Several methods in `RoslynTypeSystem` now throw `NotSupportedException` instead of returning `null`, to match the other implementations. These methods are normally not called by the name generator (or they would have failed with a `NullReferenceException` before).

### CODE REVIEW

**Comment by MrJul** [comment]
Apparently hitting a compiler bug regarding local functions and nullability - I can't reproduce locally and there's no warning either in XamlX pipelines. I'm updating the SDK here to see if that solves it.

**Comment by MrJul** [comment]
And now hitting a bug in ILRepack! See https://github.com/Gillibald/il-repack/pull/1

**Comment by maxkatz6** [comment]
@MrJul  could it be that ILRepack bug was fixed in the upstream? https://github.com/gluck/il-repack
It was recently revived and made active again

**Comment by MrJul** [comment]
> @MrJul could it be that ILRepack bug was fixed in the upstream? https://github.com/gluck/il-repack It was recently revived and made active again

Indeed, these changes are similar to mine:

https://github.com/gluck/il-repack/commit/15d48eff12c87dfd344da45076b6bae3de67a60b#diff-c49a81a215a1a411a33ee869d718122208bbfe89696ef5a091322ca938ee32e9R927-R936

Thanks for letting me know! We should probably discard the fork and use the new official version.

**Comment by MrJul** [comment]
Note: ILRepack only supported .NET Framework, so [I started porting it to .NET 8](https://github.com/MrJul/il-repack/tree/net8) a while back. While the projects compiled, I didn't had the time to update the custom NuGet pack targets yet, which was necessary to open a valid PR.

Meanwhile, ILRepack got another PR adding `dotnet tool` support: https://github.com/gluck/il-repack/pull/364
Contrary to my fork, it doesn't update the `ILRepack.Lib` to .NET 8 which I intended to use. However, I think we're completely fine with using the tool in the build tasks.

I've updated this PR to use the tool. Assemblies are now merging without errors.

File: src/tools/Avalonia.Generators/Avalonia.Generators.csproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <IncludeSymbols>false</IncludeSymbols>
     <IsRoslynComponent>true</IsRoslynComponent>
     <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
+    <Nullable>enable</Nullable>
```
Comment (maxkatz6) [inline src/tools/Avalonia.Generators/Avalonia.Generators.csproj:11]: Why not `<Import Project="..\..\..\build\NullableEnable.props" />` here?

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**